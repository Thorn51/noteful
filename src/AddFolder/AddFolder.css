.Add_folder {
  display: flex;
  flex-direction: column;
  width: 80%;
  margin: 0 auto 0 auto;
}

h2 {
  text-align: center;
  padding: 25px 0 0 0;
  font-size: 2rem;
}

.form_group {
  display: flex;
  flex-direction: column;
  margin: 10px;
}

label {
  font-size: 1.3rem;
  padding: 0px 10px 0px 10px;
}

.folder_name {
  margin: 10px;
}

.add_folder_buttons {
  display: flex;
  width: 80%;
  justify-content: space-between;
  margin: 0 auto 0 auto;
}

.form_button {
  width: 200px;
  height: 50px;
  margin: 0 0 25px 0;
  background: rgb(97, 97, 97);
  color: white;
  font-size: 1.3rem;
  cursor: pointer;
}

.form_button:disabled {
  color: lightgray;
  cursor: initial;
  font-style: oblique;
}

.error {
  font-size: 1.3rem;
  text-align: center;
  color: red;
}
