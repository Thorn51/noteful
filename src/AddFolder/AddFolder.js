import React, { Component } from "react";
import NoteContext from "../NoteContext";
import FormValidation from "../FormValidation/FormValidation";
import "./AddFolder.css";
import config from "../config";

class AddFolder extends Component {
  static contextType = NoteContext;
  constructor(props) {
    super(props);
    this.state = {
      folder_name: "",
      touched: false
    };
  }

  newFolderName(name) {
    this.setState({
      folder_name: name,
      touched: true
    });
  }

  validateName() {
    // const name = this.state.name.trim();
    if (this.state.folder_name.trim().length === 0) {
      return "A folder name is required.";
    }
  }

  onSubmit(event) {
    event.preventDefault();
    const folderName = { folder_name: this.state.folder_name };
    const fetchUrl = config.API_ENDPOINT + "/api/folders";
    const options = {
      method: "POST",
      body: JSON.stringify(folderName),
      headers: {
        "Content-Type": "application/json",
        Authorization: "Bearer " + config.API_TOKEN
      }
    };
    fetch(fetchUrl, options)
      .then(res => {
        if (!res.ok) {
          throw new Error("There has been a problem reaching the server.");
        } else {
          return res.json();
        }
      })
      .then(data => {
        this.setState({
          folder_name: "",
          touched: false
        });
        this.context.addFolder(data);
        this.props.history.goBack();
      })
      .catch(error => {
        console.log(error);
      });
  }

  cancel = () => {
    this.props.history.goBack();
  };

  render() {
    return (
      <form className="Add_folder" onSubmit={e => this.onSubmit(e)}>
        <h2>Add Folder</h2>
        <div className="form_group">
          <label htmlFor="folder_name">Folder Name (required)</label>
          <input
            type="text"
            className="folder_name"
            name="folder_name"
            id="folder_name"
            required
            onChange={e => this.newFolderName(e.target.value)}
          />
          {this.state.touched && (
            <FormValidation message={this.validateName()} />
          )}
        </div>
        <div className="add_folder_buttons">
          <button
            type="reset"
            className="form_button"
            onClick={() => this.props.history.goBack()}
          >
            Cancel
          </button>
          <button
            type="submit"
            className="form_button"
            disabled={this.validateName()}
          >
            Save
          </button>
        </div>
      </form>
    );
  }
}

export default AddFolder;
