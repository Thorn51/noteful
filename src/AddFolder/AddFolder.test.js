import React from "react";
import ReactDOM from "react-dom";
import AddFolder from "./AddFolder";
import renderer from "react-test-renderer";

describe("AddFolder", () => {
  it("renders without throwing errors", () => {
    const div = document.createElement("div");
    ReactDOM.render(<AddFolder />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it("renders the component as expected", () => {
    const tree = renderer.create(<AddFolder />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
