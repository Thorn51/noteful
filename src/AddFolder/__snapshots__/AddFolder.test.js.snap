// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFolder renders the component as expected 1`] = `
<form
  className="Add_folder"
  onSubmit={[Function]}
>
  <h2>
    Add Folder
  </h2>
  <div
    className="form_group"
  >
    <label
      htmlFor="folder_name"
    >
      Folder Name (required)
    </label>
    <input
      className="folder_name"
      id="folder_name"
      name="folder_name"
      onChange={[Function]}
      required={true}
      type="text"
    />
  </div>
  <div
    className="add_folder_buttons"
  >
    <button
      className="form_button"
      onClick={[Function]}
      type="reset"
    >
      Cancel
    </button>
    <button
      className="form_button"
      disabled="A folder name is required."
      type="submit"
    >
      Save
    </button>
  </div>
</form>
`;
